src/reduxTypes.js

```diff
@@ -298,24 +298,24 @@ export type SettingsState = {|
   streamNotification: boolean,
   browser: BrowserPreference,
   doNotMarkMessagesAsRead: boolean,
-|};
+|}>;
 
-export type StreamsState = Stream[];
+export type StreamsState = $ReadOnlyArray<Stream>;
 
-export type SubscriptionsState = Subscription[];
+export type SubscriptionsState = $ReadOnlyArray<Subscription>;
 
-export type TopicsState = {|
+export type TopicsState = $ReadOnly<{|
   [number]: Topic[],
-|};
+|}>;
 
-export type TypingState = {|
+export type TypingState = $ReadOnly<{|
   [normalizedRecipients: string]: {|
     time: number,
     userIds: UserId[],
   |},
-|};
+|}>;
```

Comment:
The inner objects here should also get read-only marks. Easy to miss those :wink:

Reply:
Ah, right. I've found a lot more of these, I think all of them. Thanks!